web/src/main/java/se/inera/intyg/webcert/web/service/utkast/dto/PreviousIntyg.java

```diff
@@ -59,11 +59,22 @@ public static PreviousIntyg of(
         final String enhetName,
         final String latestIntygsId,
         final LocalDateTime skapat) {
+        return of(sameVardgivare, sameEnhet, enableShowDoiButton, enhetName, latestIntygsId, skapat, true);
+    }
 
-        if (sameVardgivare) {
-            return new PreviousIntyg(sameVardgivare, skapat, sameEnhet, enableShowDoiButton, enhetName, latestIntygsId);
-        } else {
+    public static PreviousIntyg of(
+        final boolean sameVardgivare,
+        final boolean sameEnhet,
+        final boolean enableShowDoiButton,
+        final String enhetName,
+        final String latestIntygsId,
+        final LocalDateTime skapat,
+        final boolean shouldHideValues) {
```

Comment:
If you can, see if it is possible to give this a more explicit name to make it easier to understand without reading the whole context. Would be great if the naming is consistent when used throughout as argument to methods.

Maybe something like hideMetadataIfDifferentVardgivare.